src/GameWorldSimulator/NeoServer.Game.Creatures/Monster/Monster.cs

```diff
@@ -191,12 +200,7 @@ public void UpdateState()
         State = MonsterState.InCombat;
     }
 
-    public bool Defending { get; private set; }
-    public virtual bool IsSummon => false;
-
-    public override bool CanSeeInvisible => HasImmunity(Immunity.Invisibility); //todo: add invisibility flag
-
-    public override bool CanBeSeen => false;
+    public override bool IsThinking() => !IsSleeping;
```

Comment:
We can change this to be a property:
public override bool IsThinking {get;} => !IsSleeping;